stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/MediaAttachmentView.kt

```diff
@@ -16,17 +18,23 @@ import com.google.android.material.shape.ShapeAppearanceModel
 import io.getstream.chat.android.client.models.Attachment
 import io.getstream.chat.android.ui.R
 import io.getstream.chat.android.ui.common.extensions.internal.createStreamThemeWrapper
+import io.getstream.chat.android.ui.common.extensions.internal.displayMetrics
 import io.getstream.chat.android.ui.common.extensions.internal.dpToPx
 import io.getstream.chat.android.ui.common.extensions.internal.streamThemeInflater
 import io.getstream.chat.android.ui.common.style.setTextStyle
 import io.getstream.chat.android.ui.databinding.StreamUiMediaAttachmentViewBinding
 import io.getstream.chat.android.ui.message.list.adapter.view.MediaAttachmentViewStyle
+import kotlin.math.min
 
 internal class MediaAttachmentView : ConstraintLayout {
     var attachmentClickListener: AttachmentClickListener? = null
     var attachmentLongClickListener: AttachmentLongClickListener? = null
     var giphyBadgeEnabled: Boolean = true
 
+    private val maxMediaAttachmentWidth: Int by lazy(LazyThreadSafetyMode.NONE) {
+        (displayMetrics().widthPixels)
+    }
+
```

Comment:
This is something that I didn't find a solution for. If I wait for the views to be drawn to calculate the limit of the width based of the container of this View, then I start to have inconsistent results for views in the binding (I may get the current one of the next of the previous, accordingly to the direction of my scroll and the speed). So using methods like `doOnPreDraw` are not an option as it creates many inconsistencies in the final size of the giphy and it also creates many glitches in the scroll. 

At the moment this is a hard limit that only stops the giphy to be bigger then entire screen... Another solution would be to provide a way for customers to customize it using up of a percentage of the screen... Using percentage of the screen looks a bit odd, but dp would create problems for different sizes of screen of the final client. 

Well... this is the problem here and I'm open to tips.

Reply:
Another observation is that the problem of too big gifs may be a problem that happens so rarely that perhaps can be ignored...

Reply:
Solved using post in parent.